raiden-ts/tests/unit/epics/transport.spec.ts

```diff
@@ -1828,3 +1836,1716 @@ describe('transport epic', () => {
     });
   });
 });
+
+describe('transport epic', () => {
+  const matrixServer = 'matrix.raiden.test'; // define later in fixture constant
+  const partnerRoomId = `!partnerRoomId:${matrixServer}`;
+  const processed: Processed = {
+    type: MessageType.PROCESSED,
+    message_identifier: makeMessageId(),
+  };
+  const fetch = jest.fn(async () => ({
+    ok: true,
+    status: 200,
+    text: jest.fn(async () => `- ${matrixServer}`),
+  }));
+  Object.assign(global, { fetch });
+
+  describe('initMatrixEpic', () => {
+    test('matrix stored setup', async () => {
+      expect.assertions(2);
+      const [raiden] = await makeRaidens(1);
+      const matrix = (await raiden.deps.matrix$.toPromise()) as jest.Mocked<MatrixClient>;
+
+      const userId = matrix.getUserId()!;
+      expect(raiden.output).toContainEqual(
```

Comment:
This test was supposed to test how the `initMatrixEpic` behaves when a matrix setup is already present in state (it means the client was already initialized before). To test/simulate that, the previous version `next`ed the setup directly before subscribing to the observable, and you missed it, so this test is running a first call, instead of this intended stored setup test.

Reply:
I thought the `next`ed `matrixSetup` was the initialisation in  the previous test pattern. 
In the present pattern the
`const raiden = await makeRaiden(undefined);` 
already makes a `matrixSetup` action and after that there can no further  `matrixSetup` actions.

Reply:
`makeRaiden` will output a `matrixSetup` anyway, as it's part of its initialization, but this test wants to test how it behaves if the state already contains a populated `state.transport` setup, and this is set by that early next'd `matrixSetup`, which is missing on the new test.

Reply:
Also made the required changes but would wait for your review if anything else is required